components/brave_vpn/connection/brave_vpn_os_connection_api_base.h

```diff
@@ -100,6 +82,21 @@ class BraveVPNOSConnectionAPI
 
  private:
   friend class BraveVPNServiceTest;
+  friend class BraveVPNOSConnectionAPISim;
+  friend class BraveVPNOsConnectionAPIBaseTest;
+
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest,
+                           CreateOSVPNEntryWithValidInfoWhenConnectTest);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest,
+                           CreateOSVPNEntryWithInvalidInfoTest);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest,
+                           CheckConnectionStateAfterNetworkStateChanged);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest, HostnamesTest);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest, CancelConnectingTest);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest, ConnectionInfoTest);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest, NeedsConnectTest);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest, ConnectWithoutNetwork);
+  FRIEND_TEST_ALL_PREFIXES(BraveVPNServiceTest, OnDisconnectedWithoutNetwork);
```

Comment:
All tests mentioned here seems like can be moved from brave_vpn_service_unittests.cc to separate file in `connection` folder because they use BraveVPNService purchase calls only to initiate the connection checking. Created follow up to check and confirm this later https://github.com/brave/brave-browser/issues/27294 cc @simonhong